src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java

```diff
@@ -53,23 +53,16 @@ public static Pager extractPager(@NotNull UriInfo uriInfo) {
         final Pattern FILTER_PATTERN = Pattern.compile("filter\\[(.+)\\]");
         final String FILTER_OP = "filter:op";
 
-        String page = queryParams.getFirst(QUERY_OFFSET);
-        if (page != null) {
-            try {
-                pageBuilder.page(Integer.parseInt(page));
-            } catch (NumberFormatException nfe) {
-                throw new IllegalArgumentException(String.format(
-                        "%s expects an int but found [%s]",
-                        QUERY_OFFSET,
-                        page
-                ), nfe);
-            }
-        }
-
         String itemsPerPage = queryParams.getFirst(QUERY_LIMIT);
+        boolean usingNoLimit = false;
         if (itemsPerPage != null) {
             try {
-                pageBuilder.itemsPerPage(Integer.parseInt(itemsPerPage));
+                int limit = Integer.parseInt(itemsPerPage);
+                if (limit == Pager.NO_LIMIT) {
+                    usingNoLimit = true;
+                    pageBuilder.page(0);
```

Comment:
Ah, you do it here. It may be still more clear to have the `setFirstResult` above within the conditional.

Reply:
MMh, it's true, here I'm assuming that the default is to ignore across the PagingUtils, as is more convenient for tests. I can replicate that check also in the policies in case we decide to change this later.